nilearn/plotting/img_plotting.py

```diff
@@ -1463,3 +1463,130 @@ def plot_connectome_strength(adjacency_matrix, node_coords, node_size="auto",
         display = None
 
     return display
+
+
+def plot_carpet(img, mask_img=None, detrend=True, output_file=None,
+                figure=None, axes=None, title=None):
+    """Plot an image representation of voxel intensities across time.
+
+    This figure is also known as a "grayplot" or "Power plot".
+
+    Parameters
+    ----------
+    img : Niimg-like object
+        4D image.
+        See http://nilearn.github.io/manipulating_images/input_output.html.
+    mask_img : Niimg-like object or None, optional
+        Limit plotted voxels to those inside the provided mask (default is
+        None). If not specified a new mask will be derived from data.
+        See http://nilearn.github.io/manipulating_images/input_output.html.
+    detrend : :obj:`bool`, optional
+        Detrend the data prior to plotting (default is `True`).
+    output_file : :obj:`str` or None, optional
+        The name of an image file to which to export the plot (default is
+        None). Valid extensions are .png, .pdf, and .svg.
+        If `output_file` is not None, the plot is saved to a file, and the
+        display is closed.
+    figure : :class:`matplotlib.figure.Figure` or None, optional
+        Matplotlib figure used (default is None).
+        If None is given, a new figure is created.
+    axes : :class:`matplotlib.axes._subplots.AxesSubplot` or None, optional
+        The axes used to display the plot (default is None).
+        If None, the complete figure is used.
+    title : :obj:`str` or None, optional
+        The title displayed on the figure (default is None).
+
+    Returns
+    -------
+    figure : :class:`matplotlib.figure.Figure`
+        Figure object with carpet plot.
+
+    Notes
+    -----
+    This figure was originally developed in [1]_.
+
+    References
+    ----------
+    .. [1] Power, J. D. (2017). A simple but useful way to assess fMRI scan
+    qualities. Neuroimage, 154, 150-158. doi:
+    https://doi.org/10.1016/j.neuroimage.2016.08.009
+    """
+    img_nii = _utils.check_niimg_4d(img, dtype='auto')
+    img_data = _safe_get_data(img_nii, ensure_finite=True)
+
+    # Define TR and number of frames
+    tr = img_nii.header.get_zooms()[-1]
+    ntsteps = img_nii.shape[-1]
+
+    if not mask_img:
+        nifti_masker = NiftiMasker(mask_strategy='epi', standardize=False)
+        nifti_masker.fit(img_nii)
+        mask_data = nifti_masker.mask_img_.get_data().astype(bool)
+    else:
+        mask_nii = _utils.check_niimg_3d(mask_img, dtype='auto')
+        mask_data = _safe_get_data(mask_nii, ensure_finite=True)
+
+    data = img_data[mask_data > 0].reshape(-1, ntsteps)
+    # Detrend data
+    if detrend:
+        data = clean(data.T, t_r=tr).T
```

Comment:
this also does standardization, right ?
should be indicated somewhere

Reply:
In the original PR's review, I saw https://github.com/nilearn/nilearn/pull/1587#discussion_r172327932. Since I didn't write the original code, I'm not sure if standardization is desired (in which case I'll be more explicit), or if it's an unintended effect (in which case I can disable standardization in `clean` call).

Would be good to get @oesteban's opinion on this.

Reply:
> I'd say that standardizing before plotting is good to avoid having the eye caught by spurious details.

Okay, I be more explicit in the call and also add a comment. It should make the test easier too.